srtcore/logging.h

```diff
@@ -159,30 +160,30 @@ struct SRT_API LogDispatcher
         level(log_level),
         src_config(&config)
     {
-        // XXX stpcpy desired, but not enough portable
-        // Composing the exact prefix is not critical, so simply
-        // cut the prefix, if the length is exceeded
-
-        // See Logger::Logger; we know this has normally 2 characters,
-        // except !!FATAL!!, which has 9. Still less than 32.
-        // If the size of the FA name together with severity exceeds the size,
-        // just skip the former.
-        if (logger_pfx && strlen(prefix) + strlen(logger_pfx) + 1 < MAX_PREFIX_SIZE)
+        size_t your_pfx_len = your_pfx ? strlen(your_pfx) : 0;
```

Comment:
This avoids using snprintf and playing with alternatives. The length of the strings is taken once and then all copying is done on memcpy with known size. The snprintf function was prone to various warnings either through clang or MSVC.